### DISCUSSION TITLE
[firebase_crashlytics] Support for deobfuscation of Dart reports

### DISCUSSION THREAD

**Discussion by kroikie** [firebase/flutterfire#5562] [Feature request]
I can't even find any info on how to find dart obfuscation maps to deobfuscate reports manually

**Comment by kroikie** [comment]
@rostopira

The issue at https://github.com/flutter/flutter/issues/30303 has been closed and moved here. Future collaboration on this issue will be done here.

**Comment by sperochon** [comment]
Hello,

I have created a web flutter app to do dart deobfuscation easily:
https://www.mobiquite.fr/dart-flutter-stacktrace-deobfuscator/

It is inspired from the script of @rostopira
https://github.com/flutter/flutter/issues/30303#issuecomment-522544995

**Comment by kikuchy** [comment]
Any update?

**Comment by rostopira** [comment]
@kikuchy yep, update: https://flutter.dev/docs/deployment/obfuscate
@kroikie Please, close this issue, it's resolved

**Reply by 4ntoine** [reply]
it's probably outdated. i don't have `/out/android/` after neither `flutter build apk -h` nor `flutter build appbundle`

**Comment by rostopira** [comment]
Oops, missed that issue is about crashlytics, not only obfuscation, should not be closed

**Comment by lsuhov** [comment]
Hello,
Please prioritize the deobfuscation of dart stack traces in Crashlytics. How are the production Flutter apps managing this part? I doubt that `flutter symbolize` can be considered a reasonable solution if you have multiple exceptions to check.

Ideally the complete solution should be offered with firebase_app_distribution from fastlane.

**Comment by 2math** [comment]
Best solution is to be able to upload the symbols per version in firebase and crashlitics to be able to translate the exception.
Something like mappings on Play Store

**Comment by singun-lxd** [comment]
Any solutions?

**Comment by talent-apps** [comment]
Any update?
At the moment, the `firebase_crashlytics` plugin is useless for apps that use Dart obfuscation.
Does this plugin have any owners? Why such a crucial issue is left unaddressed?

**Comment by nt4f04uNd** [comment]
@talent-apps it's not, you still can manually deobfuscate crash reports

**Comment by dluksza** [comment]
@nt4f04uNd the first step would be to include this information in the README and document how to do this manually. I've tried deobfuscating my crash reports manually but it never worked for me.

**Comment by dayron9110** [comment]
Any updates?

**Comment by dayron9110** [comment]
News?

**Reply by MichalDanielDobrzanski** [reply]
Shame that you maintainers have no interest in such an important, security-based topic.

**Comment by gwennguihal** [comment]
Any update ?

**Comment by csells** [comment]
@Ehesp can Invertase take a look at this?

**Comment by Salakar** [comment]
Hey folks,

I don't think this is something FlutterFire itself can solve since obfuscation and de-obfuscation are not something a Flutter plugin would have any control over (?) - but this maybe something we can document on how to do it - though I'm not sure what more we can add that the official flutter obfuscation docs don't already cover? (please let me know).

The FlutterFire Crashlytics plugin still reports Dart stack traces to Firebase, whether its obfuscated or not - so once they're reported it should be a case of extracting it from the Firebase Console and de-obfuscating it via the existing Flutter tooling. Ideally not having to do this yourself and having the stack traces automatically get symbolized on the Firebase Console would be nice - but this is something we as plugin maintainers currently don't have any control over unfortunately.

**Comment by gwennguihal** [comment]
Hi, thank you for your answer. If I understand, it's a firebase feature?

**Reply by Salakar** [reply]
Correct

**Comment by vanlooverenkoen** [comment]
Indeed this is a feature that should be implemented by firebase itself. @Salakar do you know where we can log that? Or do you know if Google/Firebase is already working on this?

**Comment by vanlooverenkoen** [comment]
I created a firebase support ticket. And this was the response: 

----

Just to set some expectations, Flutter is considered as one of the [external frameworks](https://firebase.google.com/docs/libraries#framework-libraries) and not covered by [Firebase support](https://firebase.google.com/support/scope) but I'd still love to point you to the right direction. Per checking, we have an existing feature request for native Flutter crash reporting and I would love to add this case as an upvote on having this feature. However, I can't share any details or timelines on this at the moment. For now, please keep an eye on our [release notes](https://firebase.google.com/support/releases) and official [blog posts](https://firebase.googleblog.com/) for any new features, bug fixes or improvements.

----

Looking at this answer I am not very hopeful this will work in the coming months, or even years. @Salakar do you have more information on this?

**Comment by suesitran** [comment]
any update?

**Comment by suesitran** [comment]
update on my side: my observation is that Crashlytic deobfuscate stacktrace for fatal error correctly, but not for Exception. We need both to be deobfuscated.

**Comment by andychan06** [comment]
any update?

**Comment by vanlooverenkoen** [comment]
Now that flutter has "graduated" to become a first-class and fully supported part of the firebase offering. 

When will deobfuscated stack traces be added? I don't see anything in the documentation, @Salakar do you have any idea when this will be implemented?

**Comment by 4ntoine** [comment]
Any update on this? Related question on SO https://stackoverflow.com/questions/72725168/how-to-deobfuscate-java-stacktrace-in-flutter-app-with-firebase-crashlytics

**Comment by aprzedecki** [comment]
it works without big issues for me, as soon as you have the symbols file you can just copy-paste to some file the stack you have under Stack trace => TXT and this command deobfuscates the stack:

```
 flutter symbolize -i <stack trace file> -d app.symbols
```

**Reply by vanlooverenkoen** [reply]
yes offcourse that works. But that is not something we should be doing manually. 
When using kotlin the every crash is deobfuscated. so there should be a way to upload the dart/flutter symbols files as well to crashlytics firebase

**Comment by aprzedecki** [comment]
I totally agree. Would be a time saving feature.

**Comment by zmeid** [comment]
Isn't that the solution?(only for android):
https://firebase.google.com/docs/crashlytics/get-started?platform=flutter#add-sdk

<img width="881" alt="image" src="https://user-images.githubusercontent.com/13780749/233944559-6f421ab4-4bce-4cb3-af1e-da8529c92653.png">

**Reply by lsuhov** [reply]
We tried this already and it is not working, even though no error is displayed after upload of symbols.

**Reply by duweiwang** [reply]
It works for me! I upload symbols with above command after jenkins post action.

**Comment by Rishyash** [comment]
https://medium.com/@rishyash8/how-to-read-an-obfuscated-stack-trace-in-firebase-crashlytics-in-flutter-3a189dd5abc3
